Objects/longobject.c

```diff
@@ -220,11 +221,14 @@ _PyLong_FromMedium(sdigit x)
 {
     assert(!IS_SMALL_INT(x));
     assert(is_medium_int(x));
-    /* We could use a freelist here */
-    PyLongObject *v = PyObject_Malloc(sizeof(PyLongObject));
+
+    PyLongObject *v = _Py_FREELIST_POP(PyLongObject, ints);
```

Comment:
To fix the refleak, use this instead:
```suggestion
    PyLongObject *v = _Py_FREELIST_POP_MEM(PyLongObject, ints);
```

Reply:
You can add a comment saying that this is due to `_Py_FREELIST_POP` calling `_Py_NewReference`, which is incompatible with the small int cache.

Reply:
If I understand correctly, we have to call `_Py_FREELIST_POP_MEM` instead of `_Py_FREELIST_POP` because the new reference is created later in `_PyObject_Init`.

The comment about being incompatible with the small int cache I do not completely understand.